src/utils/fetchWaiting.ts

```diff
@@ -0,0 +1,8 @@
+export default async function fetchWaiting(input: RequestInfo, init?: RequestInit, minimumLoadingTime = 0) {
+  const startTime = Date.now()
+  const response = await fetch(input, init)
+  const endTime = Date.now()
+  const timeoutTime = minimumLoadingTime - (endTime - startTime)
+  await new Promise(resolve => setTimeout(resolve, timeoutTime))
+  return response
```

Comment:
another way to achieve is create a `setTimeout` with `minimumLoadingTime`, then `Promise.all` it with `fetch`. This solution is much straighforward IMO, dont need to calc time left.

Reply:
awesome

Reply:
Agree